Source/ExecutionEngine/ExecutionEngine.cs

```diff
@@ -706,21 +706,17 @@ private Implementation[] GetPrioritizedImplementations(Program program)
       }
 
       await verifyImplementationSemaphore.WaitAsync(cts.Token);
-      try {
-
-        ImplIdToCancellationTokenSource.AddOrUpdate(id, cts, (k, ov) => cts);
+      ImplIdToCancellationTokenSource.AddOrUpdate(id, cts, (k, ov) => cts);
 
-        var coreTask = Task.Run(() => VerifyImplementation(processedProgram, stats, er, cts.Token,
-          implementation,
-          programId, taskWriter), cts.Token);
+      var coreTask = Task.Run(() => VerifyImplementation(processedProgram, stats, er, cts.Token,
+        implementation,
+        programId, taskWriter), cts.Token);
 
-        var verificationResult = await coreTask;
-        return verificationResult;
-      }
-      finally {
+      var _ = coreTask.ContinueWith(t => {
```

Comment:
I think it will. The callback in `ContinueWith` is called if the Task goes to the completed state, either successfully, cancelled or if an exception occurred.